processing/src/main/java/io/druid/query/filter/Filter.java

```diff
@@ -26,7 +26,51 @@
  */
 public interface Filter
 {
-  public ImmutableBitmap getBitmapIndex(BitmapIndexSelector selector);
-  public ValueMatcher makeMatcher(ValueMatcherFactory factory);
-  public ValueMatcher makeMatcher(ColumnSelectorFactory columnSelectorFactory);
+  // this should be supported, at least
+  ValueMatcher makeMatcher(ValueMatcherFactory factory);
+
+  // optional. bitmap based filter has priority over value matcher below
+  ImmutableBitmap getBitmapIndex(BitmapIndexSelector selector);
+
+  // optional
+  ValueMatcher makeMatcher(ColumnSelectorFactory columnSelectorFactory);
+
+  boolean supportsBitmap();
+
+  boolean supportsValueMatcher();
+
+  // extend this
+  abstract class AbstractFilter implements Filter
+  {
+    @Override
+    public ImmutableBitmap getBitmapIndex(final BitmapIndexSelector selector)
+    {
+      throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public ValueMatcher makeMatcher(ColumnSelectorFactory columnSelectorFactory)
+    {
+      throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public boolean supportsBitmap()
+    {
+      return false;
+    }
+
+    @Override
+    public boolean supportsValueMatcher()
+    {
+      return false;
+    }
+
+    @Override
+    public boolean equals(Object other)
+    {
+      // not for serious usage
```

Comment:
I've used this to check equality of filter in UT. If it does not look good, I'll think another way to do it.